jetty-home/src/main/resources/bin/jetty.sh

```diff
@@ -118,35 +118,86 @@ findDirectory()
   done
 }
 
+# test if process specified in PID file is still running
 running()
 {
-  if [ -f "$1" ]
-  then
-    local PID=$(cat "$1" 2>/dev/null) || return 1
-    kill -0 "$PID" 2>/dev/null
-    return
+  local PIDFILE=$1
+  if [ -r "$PIDFILE" ] ; then
+    local PID=$(tail -1 "$PIDFILE")
+    if kill -0 "$PID" 2>/dev/null ; then
+      return 0
+    fi
   fi
-  rm -f "$1"
   return 1
 }
 
+# Test state file (after timeout) for started state
 started()
 {
-  # wait for 60s to see "STARTED" in PID file, needs jetty-started.xml as argument
-  for ((T = 0; T < $(($3 / 4)); T++))
+  local STATEFILE=$1
+  local PIDFILE=$2
+  local STARTTIMEOUT=$3
+  # wait till timeout to see "STARTED" in state file, needs --module=state as argument
+  for ((T = 0; T < $(($STARTTIMEOUT / 4)); T++))
   do
     sleep 4
-    [ -z "$(tail -1 $1 | grep STARTED 2>/dev/null)" ] || return 0
-    [ -z "$(tail -1 $1 | grep STOPPED 2>/dev/null)" ] || return 1
-    [ -z "$(tail -1 $1 | grep FAILED 2>/dev/null)" ] || return 1
-    local PID=$(cat "$2" 2>/dev/null) || return 1
-    kill -0 "$PID" 2>/dev/null || return 1
-    echo -n ". "
+    if [ -r $STATEFILE ] ; then
+      STATENOW=$(tail -1 $STATEFILE)
+      (( DEBUG )) && echo "State (now): $STATENOW"
+      case "$STATENOW" in
+        STARTED*)
+          return 0;;
+        STOPPED*)
+          return 1;;
+        FAILED*)
+          return 1;;
+      esac
+      echo -n "-"
+    else
+      (( DEBUG )) && echo "Unable to read State File: $STATEFILE"
+      echo -n ":"
```

Comment:
Is this echo in the right place?
Also, should not this branch return 1, as the file is not readable?

Reply:
We don't return here.
As the state file might be unreachable at first, but then show up in later iteration of the loop.